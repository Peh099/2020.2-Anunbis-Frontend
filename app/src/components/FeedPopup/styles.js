import styled from 'styled-components';

export const Container = styled.div`
  width: 372px;
  height: fit-content;
  background-color: #fffde7;
  border-bottom-right-radius: 2%;
  border-bottom-left-radius: 2%;
  display: flex;
  flex-direction: column;
  @media (max-width: 460px) {
    font-size: 13px;
    width: 80vw;
  }
`;

export const Header = styled.div`
  height: 20px;
  width: inherit;
  background-color: #212121;
  display: flex;
  justify-content: center;

  Button {
    margin-inline: 0px;
    color: #fffde7;
    border: 0px;
    box-shadow: none;
  }
`;

export const Title = styled.div`
  display: flex;
  color: #fffde7;
  margin-left: auto;
  margin-right: auto;
  font-family: 'Comfortaa', cursive;
`;

export const Content = styled.main`
  align-self: center;
  justify-content: center;
  margin-bottom: 10px;
  @media (max-width: 460px) {
    width: 100%;
  }
`;
