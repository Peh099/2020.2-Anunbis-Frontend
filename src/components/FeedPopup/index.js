import styled from 'styled-components';
import Button from '../Button';

const Container = styled.div`
    width: 372px;
    min-height: 380px;
    height: fit-content;
    background-color: #FFFDE7;
    border-bottom-right-radius: 2%;
    border-bottom-left-radius: 2%;
    display: flex;
    flex-direction: column;

`;

const Header = styled.div`
    height: 20px;
    width: inherit;
    background-color: #212121;
    display: flex;
    justify-content: center;

    Button {
        margin-inline: 0px;
        color: #FFFDE7;
        border: 0px;
    }
`;

const Title = styled.div`
    width: 50px;
    display:flex;
    color: #FFFDE7;
    margin-left: auto;
    margin-right: auto;
`;

const Content = styled.main`
align-self: center;
justify-content: center;
`;

export default function FeedPopup({close, title, children}) {

    return (
        <Container>
            <Header>
                <Title>{title}</Title>
                <Button onClick={close} text="X" radius="0px" padding="2px 5px" backColor="#F44336" />
            </Header>
            <Content>
                {children}
            </Content>

        </Container>
    );
}