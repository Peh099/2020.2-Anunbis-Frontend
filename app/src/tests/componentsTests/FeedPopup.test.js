import renderer from 'react-test-renderer';
import FeedPopup from '../../components/FeedPopup';

describe('Snapshot FeedPopup component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<FeedPopup />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
