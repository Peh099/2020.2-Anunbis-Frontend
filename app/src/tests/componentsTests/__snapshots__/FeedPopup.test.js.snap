// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot FeedPopup component matches the snapshot 1`] = `
<div
  className="sc-gtsrHT guyQyB"
>
  <div
    className="sc-dlnjwi gKbNGq"
  >
    <div
      className="sc-hKFxyN bhTXYs"
    />
    <button
      className="sc-bdnxRM jAGuyW"
      radius="0px"
    >
      X
    </button>
  </div>
  <main
    className="sc-eCApnc bQqanu"
  />
</div>
`;
